activation/handler_v2.go

```diff
@@ -630,100 +625,87 @@ func (h *HandlerV2) syntacticallyValidateDeps(
 					zap.Stringer("id", atx.ID()),
 					zap.Int("index", invalidIdx.Index),
 				)
-				// TODO generate malfeasance proof
+				// TODO(mafa): finish proof
+				proof := &wire.ATXProof{
+					ProofType: wire.InvalidPost,
+				}
+				if err := h.malPublisher.Publish(ctx, id, codec.MustEncode(proof)); err != nil {
+					return nil, fmt.Errorf("publishing malfeasance proof for invalid post: %w", err)
+				}
 			}
 			if err != nil {
-				return nil, nil, fmt.Errorf("validating post for ID %s: %w", id.ShortString(), err)
+				return nil, fmt.Errorf("validating post for ID %s: %w", id.ShortString(), err)
 			}
 			parts.units[id] = post.NumUnits
 		}
 	}
 
 	if atx.Initial == nil {
 		if smesherCommitment == nil {
-			return nil, nil, errors.New("ATX signer not present in merged ATX")
+			return nil, errors.New("ATX signer not present in merged ATX")
 		}
 		err := h.nipostValidator.VRFNonceV2(atx.SmesherID, *smesherCommitment, atx.VRFNonce, atx.TotalNumUnits())
 		if err != nil {
-			return nil, nil, fmt.Errorf("validating VRF nonce: %w", err)
+			return nil, fmt.Errorf("validating VRF nonce: %w", err)
 		}
 	}
 
 	parts.ticks = nipostSizes.minTicks()
-
-	return &parts, nil, nil
+	return &parts, nil
 }
 
 func (h *HandlerV2) checkMalicious(
+	ctx context.Context,
 	tx *sql.Tx,
 	watx *wire.ActivationTxV2,
 	marrying []marriage,
-) (bool, *mwire.MalfeasanceProof, error) {
+) error {
 	malicious, err := identities.IsMalicious(tx, watx.SmesherID)
 	if err != nil {
-		return false, nil, fmt.Errorf("checking if node is malicious: %w", err)
+		return fmt.Errorf("checking if node is malicious: %w", err)
 	}
 	if malicious {
-		return true, nil, nil
+		return nil
 	}
 
-	proof, err := h.checkDoubleMarry(tx, marrying)
-	if err != nil {
-		return false, nil, fmt.Errorf("checking double marry: %w", err)
-	}
-	if proof != nil {
-		return true, proof, nil
+	if err := h.checkDoubleMarry(ctx, tx, marrying); err != nil {
+		return fmt.Errorf("checking double marry: %w", err)
 	}
```

Comment:
Previously, it would stop checking for other malfeasance types immediately after detecting malfeasance (the `if proof != nil { return ... }` check). There is no point in continuing once malfeasance is found. How about returning a bool meaning "identified malfeasance"? Something like:

```go
malicious, err := h.checkDoubleMarry(ctx, tx, marrying)
if err != nil {
	return fmt.Errorf("checking double marry: %w", err)
}
if malicious {
	return nil
}
```